source/core/Variants.cpp

```diff
@@ -55,14 +55,23 @@ VIREO_FUNCTION_SIGNATURET(VariantToData, VariantToDataParamBlock)
 {
     ErrorCluster *errPtr = _ParamPointer(ErrorClust);
     if (!errPtr || !errPtr->status) {
-        TypeRef variant = _Param(Variant);
+        TypeRef inputType = _ParamImmediate(InputData._paramType);
+        void* inputData = _ParamImmediate(InputData)._pData;
+
         TypeRef destType = _ParamImmediate(DestData._paramType);
+        void* destData = _ParamImmediate(DestData)._pData;
 
-        if (variant->IsA(destType)) {
-            void* destData = _ParamImmediate(DestData)._pData;
-            variant->CopyData(variant->Begin(kPARead), destData);
+        if (inputType->Name().Compare(&TypeCommon::TypeVariant)) {
+            TypeRef variantInnerType = *(TypeRef *)_ParamImmediate(InputData._pData);
+            if (variantInnerType->IsA(destType)) {
+                variantInnerType->CopyData(variantInnerType->Begin(kPARead), destData);
+            } else if (errPtr) {
+                errPtr->SetErrorAndAppendCallChain(true, 91, "Variant To Data");
```

Comment:
Can we use error code consts instead of hardcoding the number directly?

Reply:
Same for "Variant To Data" string.

Reply:
Definitely yes. Since Sid is out, I'll take care of this on my PR.